### PR TITLE
Achievement list refactor and breadcrumbs

### PR DESCRIPTION
**Pull request by Tschonti**
- AchievementList.tsx -> AchievementCategoryList.tsx
- AchievementCategoryItem.tsx -> AchievementCategoryPage.tsx
- Achievement categories and challenges both use the list from RiddleList
- More subtle colors on hover (if you like them it should be changed on the riddle page as well)
- Status badges are now solid because the transparent style didn't look good with grey background
- Created a reusable Breadcrumb component using Chakra
- There's an extra API call in AchievementPage just to get the category name for the breadcrumb, this might be simplified later on the backend

### CODE REVIEW

File: src/main/client/src/components/@pages/AchievementPage.tsx [hunk 1]
```diff
@@ -144,8 +155,23 @@ export const AchievementPage: React.FC = (props) => {
...
+    {
+      title: achDetails.achievement?.title
+    }
+  ]
+
   return (
     <Page {...props} loginRequired>
+      <CustomBreadcrumb items={breadcrumbItems} />
       <Heading mb={0}>{achDetails.achievement?.title}</Heading>
       <AchievementStatusBadge status={achDetails.status} fontSize="lg" />
```
Comment (berenteb) [inline src/main/client/src/components/@pages/AchievementPage.tsx:176]: Top margin of 5

File: src/main/client/src/components/@pages/AchievementPage.tsx [hunk 2]
```diff
@@ -144,8 +155,23 @@ export const AchievementPage: React.FC = (props) => {
...
       <Paragraph mt={2}>{achDetails.achievement?.description}</Paragraph>
```
Comment (berenteb) [inline src/main/client/src/components/@pages/AchievementPage.tsx:177]: Top margin of 5

**Review by berenteb — CHANGES_REQUESTED** [review]
Those breadcrumbs are dope AF, nice feature, we must implement this on other pages!
I am missing the margins on the AchievementPage, you didn't like it or just missed that? I was thinking about a mt={5} as discussed.
Képernyőfotó 2022-01-24 - 21 18 18
?

File: src/main/client/src/components/@pages/AchievementPage.tsx (hunk 1 above)
Comment (Tschonti) [reply src/main/client/src/components/@pages/AchievementPage.tsx:176]: I forgot about the margins this time and I also forgot to check the uploaded file size before submission, so I'll come back to this tomorrow.

**Pull request merged**